epick_description/launch/epick_gripper.launch.py

```diff
@@ -26,27 +26,33 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+"""Launch file to start the Epick gripper."""
+
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+
+import xacro
+
 from launch import LaunchDescription
 from launch.actions import (
     DeclareLaunchArgument,
-    IncludeLaunchDescription,
     OpaqueFunction,
-    RegisterEventHandler,
 )
-from launch.conditions import IfCondition
-from launch.event_handlers import OnProcessExit
-from launch.launch_description_sources import PythonLaunchDescriptionSource
 from launch.substitutions import (
     LaunchConfiguration,
     PathJoinSubstitution,
 )
-from launch_ros.actions import Node
-from launch_ros.substitutions import FindPackageShare
-
-import xacro
 
 
 def launch_setup(context, *args, **kwargs):
+    """
+    Launch function.
+
+    Declare all parameters and extract their values, then start the nodes.
+    'param: context' is the context object passed in by the launch framework.
```

Comment:
Curious if this follows a documentation convention and where to find it.

Reply:
This uses something called opaque function, at the bottom of the script. Using this technique you can access the value of a parameter before passing it down to a node. This allows for debugging, but also for processing the content of a parameter (e.g. clamping) before sending it to a node.

Reply:
Thanks for the explanation!